### PR TITLE
Make top level command backwards compatible for now. 

### PR DESCRIPTION
**Pull request by kyle-rader**
To allow for a smoother update for your customers, we can prevent the new command structure from breaking existing usage by duplicating the logic on both the `CommandAzureAUth` and the new `CommandAad`. 

This is simple to do by inheriting from `CommandAad`. 

The plan will be to remove this, once we have proved with telemetry, that usage has moved over to use the `aad` command.

## Example Back Compat
```
microsoft-authentication-cli - backwards-compat via .NET v7.0.102
❯ .\bin\azureauth.cmd --alias ado
Token cache warm for <kyle>

microsoft-authentication-cli on - backwards-compat via .NET v7.0.102 took 4s
❯ .\bin\azureauth.cmd aad --alias ado
Token cache warm for <kyle>
```

## Help
It's colored
image

```
.\bin\azureauth.cmd --help
1.0.0.0

A CLI interface to MSAL (Microsoft Authentication Library).

⚠️ NOTICE: Use `azureauth aad [options]`.
⚠️ The top-level azureauth command is deprecated and will print help text soon.

-- azureauth [options]
++ azureauth aad [options]


Usage: azureauth [command] [options]

Options:
  --resource     The ID of the resource you are authenticating to.
  --client       The ID of the App registration you are authenticating as.
  --tenant       The ID of the Tenant where the client and resource entities exist in
  --prompt-hint  The prompt hint text for WAM prompts and web mode.
  --scope        Scopes to request. By default, the only scope requested is <resource ID>\.default.
                 Passing in one or more values here will override the default.
  --clear        Clear the token cache for this AAD Application.

  --domain       Preferred domain to filter cached accounts by. If a single account matching the preferred domain is in the cache it is used, otherwise an account picker will be launched.

  --mode         Authentication mode. Default: iwa (Integrated Windows Auth), then broker, then web.
                 You can use any combination of modes with multiple instances of the --mode flag.
                 Allowed values: [all, iwa, broker, web, devicecode]
  --output       Output mode. Controls how the token information is printed to stdout.
                 Default: status (only print status, no token).
                 Available: [status, token, json, none]

  --alias        The name of an alias for config options loaded from the config file.
  --config       The path to a configuration file.
  --timeout      The number of minutes before authentication times out.
                 Default: 10 minutes.
  -?|-h|--help   Show help information.
  --version      Show version information.
  --debug        Turn off telemetry reporting and enable trace logging
  --verbosity    Configure the minimum log level to show.
                 Allowed values are: trace, debug, info, information, warn, warning, error, critical, fatal, none, off.

Commands:
  aad            todo
  ado            A collection of Azure Devops (ADO) specific authentication commands.
  info           Shows AzureAuth debug information. Please provide when asking for help.

Run 'azureauth [command] -?|-h|--help' for more information about a command.
```

And
```
.\bin\azureauth.cmd aad --help
1.0.0.0

todo

Usage: azureauth aad [options]

Options:
  --resource     The ID of the resource you are authenticating to.
  --client       The ID of the App registration you are authenticating as.
  --tenant       The ID of the Tenant where the client and resource entities exist in
  --prompt-hint  The prompt hint text for WAM prompts and web mode.
  --scope        Scopes to request. By default, the only scope requested is <resource ID>\.default.
                 Passing in one or more values here will override the default.
  --clear        Clear the token cache for this AAD Application.

  --domain       Preferred domain to filter cached accounts by. If a single account matching the preferred domain is in the cache it is used, otherwise an account picker will be launched.

  --mode         Authentication mode. Default: iwa (Integrated Windows Auth), then broker, then web.
                 You can use any combination of modes with multiple instances of the --mode flag.
                 Allowed values: [all, iwa, broker, web, devicecode]
  --output       Output mode. Controls how the token information is printed to stdout.
                 Default: status (only print status, no token).
                 Available: [status, token, json, none]

  --alias        The name of an alias for config options loaded from the config file.
  --config       The path to a configuration file.
  --timeout      The number of minutes before authentication times out.
                 Default: 10 minutes.
  -?|-h|--help   Show help information.
  --version      Show version information.
  --debug        Turn off telemetry reporting and enable trace logging
  --verbosity    Configure the minimum log level to show.
                 Allowed values are: trace, debug, info, information, warn, warning, error, critical, fatal, none, off.
```

### CODE REVIEW

File: src/AzureAuth/Commands/CommandAzureAuth.cs [hunk 1]
```diff
@@ -3,33 +3,32 @@
...
     /// <summary>
     /// The command main class parses commands and dispatches to the corresponding methods.
     /// </summary>
-    [Command(Name = "azureauth", Description = "A CLI interface to MSAL (Microsoft Authentication Library)")]
+    [Command(Name = "azureauth", Description = "A CLI interface to MSAL (Microsoft Authentication Library)." +
+        "\n\n⚠️ NOTICE: This top level command has moved!" +
+        "\n\nCalling 'azureauth' with no commands in future versions will print the avaialble commands instead of performing authentication." +
+        "\n\n ❌ azureauth [options]" +
+        "\n              ⬇️⬇️" +
+        "\n ✅ azureauth aad [options]\n")]
```
Comment (reillysiemens) [inline src/AzureAuth/Commands/CommandAzureAuth.cs:20]: Personally, I think it's sufficient to keep it brief and say
```
⚠️ NOTICE: Use `azureauth aad [options]`. The top-level azureauth command will be deprecated soon.
```

**Comment by goagain** [comment]
I would like to have the following if `LogWarning` is outputted by stderr.
```csharp
        public new int OnExecute()
        {
            this.logger.LogWarning("deprecated message...");
            return base.OnExecute();
        }
```

**Comment by kyle-rader** [comment]
> I would like to have the following if `LogWarning` is outputted by stderr.
> 
> ```cs
>         public new int OnExecute()
>         {
>             this.logger.LogWarning("deprecated message...");
>             return base.OnExecute();
>         }
> ```

While that may be nice in some scenarios to help get the word out, it could also introduce a breaking change when we are explcitly saying we are not breaking it. Since many things shell out and _can_ (I am not saying this is correct, or good), but can interrpret stderr as something failed and mess up existing behavior. So I'd like to avoid that, even though, technically, that kind of abreak is on the consumer for treating stderr as an exit code.

**Review by reillysiemens — APPROVED** [review]

**Review by mvanchaa — APPROVED** [review]

**Review by goagain — APPROVED** [review]
We need a label for this PR. I am not sure it should be an `enhancement` or `documentation`

**Review by shalinikhare27 — APPROVED** [review]

**Pull request merged**